RoFILib/configuration/Configuration.h

```diff
@@ -236,16 +274,60 @@ class Configuration
 
     bool isValid();
 
-    // Fill in the rotation attribute according to the first fixed module.
+    /**
+     * \brief Computes matrices for all shoes.
+     * 
+     * This method is called in method isValid.
+     * 
+     * User should call this if they create new configuration or they change 
+     * some angles and they don't call method isValid.
+     * 
+     * \return `false` if there is inconsistency in computing matrices,
+     * i.e. matrix for one shoe was computed two different ways and 
+     * the results aren't equal
+     */
     bool computeMatrices();
+    
+    /**
+     * \brief Computes matrix for shoe at the end of \p edge from shoe at the
+     * begining of \p edge.
+     * 
+     * For example, module with id = 1 is connected from shoe A to shoe B of 
+     * module with id = 2. It takes matrix of shoe A of module 1 and computes
+     * matrix of shoe B of module 2 from that.
+     * 
+     * If neither matrix of shoes of the first module is computed, error is
+     * thrown.
+     * 
+     * \return Matrix of the shoe connected to the end of \p edge
+     */
     Matrix computeConnectedMatrix(Edge edge) const;
 
+    /**
+     * \brief Given \p id and \p side computes matrix for the other shoe 
+     * of the given module.
+     * 
+     * \return Matrix of the other shoe of the given module
+     */
+    Matrix computeOtherSideMatrix(ID id, ShoeId side) const;
+
     bool connected() const;
 
     bool collisionFree() const;
 
     Vector massCenter() const;
 
+    /**
+     * \brief Tries to execute given \p action.
+     * 
+     * `False` is returned if either you try to add edge to already occupied
+     * dock, erase non-existent edge or if `rotate_joint` return `False`.
```

Comment:
Do not refer to implementation details (a reference to \rotate_joint`) - better state "perform invalid rotation".